sdks/java/io/amazon-web-services2/src/main/java/org/apache/beam/sdk/io/aws2/kinesis/enhancedfanout/EFOShardSubscribersPool.java

```diff
@@ -330,11 +352,12 @@ public KinesisReaderCheckpoint getCheckpointMark() {
     return new KinesisReaderCheckpoint(checkpoints);
   }
 
-  public boolean stop() throws InterruptedException {
+  public boolean stop() {
     LOG.info("Stopping pool {}", poolId);
+    isStopped = true;
     state.forEach((shardId, st) -> st.subscriber.cancel());
-    scheduler.shutdown();
-    return scheduler.awaitTermination(SCHEDULER_SHUTDOWN_TIMEOUT_MS, MILLISECONDS);
+    scheduler.shutdownNow(); // immediately discard all scheduled tasks
```

Comment:
We don't want to wait for any scheduled re-subscription. I think it's totally fine to shutdown immediately.
Thinking about this , we should even moving the above line below and cancel subscribers afterwards to make sure there 's not a race condition where we've just re-subscribed between cancelling and shutting the scheduler down.